pydoctor/test/test_commandline.py

```diff
@@ -59,3 +63,45 @@ def test_cache_disabled_by_default() -> None:
     parser = driver.getparser()
     (options, _) = parser.parse_args([])
     assert not options.enable_intersphinx_cache
+
+
+def test_cli_warnings_on_error() -> None:
+    """
+    The -W option is disabled by default.
+    """
+    options, args = driver.parse_args([])
+    assert options.warnings_as_errors == False
+
+    options, args = driver.parse_args(['-W'])
+    assert options.warnings_as_errors == True
```

Comment:
I'm not sure we need a test case for this, especially since the other two test cases already check the behavior with and without the `-W` command line argument.

Reply:
I have created this tests as an unit test for the CLI arguments parsing.
The other tests are for CLI behaviour.

But with the introduction of the long format, I can reuse this test for the long form.

Reply:
The argument parsing is done by the standard library though, so what we're testing is that `default=False` is present in the option's definition and that `optparse` works as documented. It just seems like a lot of effort to test something low-risk.

But as it's already written and I don't expect this to require a lot of maintenance, it's fine to keep it.

Reply:
This test is more about the fact that we have defined the  `--warnings-as-errors` ... I will keep it.

Since this test is easy to write, I prefer to have them as later it might help detect some strange side effects that we might introduce with "advances" command line parsing options.

So it's just a simple regression  tests since we don't use `--warnings-as-errors` in the other tests.